databricks/koalas/groupby.py

```diff
@@ -2177,7 +2177,10 @@ def _resolve_grouping(kdf: DataFrame, by: List[Union[Series, Tuple[str, ...]]])
         new_by_series = []
         for col_or_s in by:
             if isinstance(col_or_s, Series):
-                new_by_series.append(col_or_s)
+                if col_or_s._kdf is kdf:
+                    new_by_series.append(col_or_s)
+                else:
+                    new_by_series.append(col_or_s.rename(col_or_s.name))
```

Comment:
Shall we add a comment that the rename is being added because two different DataFrames are joined and the original labels in the name can be lost (is it correct?)